devicetypes/smartthings/zigbee-dual-switch.src/zigbee-dual-switch.groovy

```diff
@@ -0,0 +1,140 @@
+*  in compliance with the License. You may obtain a copy of the License at:
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+*  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+*  for the specific language governing permissions and limitations under the License.
+*
+*/
+
+metadata {
+    definition (name: "ZigBee Multi Switch", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "oic.d.switch", runLocally: false, minHubCoreVersion: '000.019.00012', executeCommandsLocally: true, vid: "generic-switch" ) {
+        capability "Actuator"
+        capability "Configuration"
+        capability "Refresh"
+        capability "Health Check"
+        capability "Switch"
+        
+        command "childOn", ["string"]
+        command "childOff", ["string"]
+        
+        fingerprint  profileId: "0104", inClusters: "0000, 0005, 0004, 0006", manufacturer: "0000000", model: "cc891c853291418daa4e26cb315a3a0c", deviceJoinName: "Orvibo Smart Switch 1"
+       }
+    // simulator metadata
+    simulator {
+        // status messages
+        status "on": "on/off: 1"
+        status "off": "on/off: 0"
+
+        // reply messages
+        reply "zcl on-off on": "on/off: 1"
+        reply "zcl on-off off": "on/off: 0"
+    }
+
+    tiles(scale: 2) {
+        multiAttributeTile(name:"switch", type: "lighting", width: 6, height: 4, canChangeIcon: true){
+            tileAttribute ("device.switch", key: "PRIMARY_CONTROL") {
+                attributeState "on", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+                attributeState "off", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+                attributeState "turningOn", label:'${name}', action:"switch.off", icon:"st.switches.light.on", backgroundColor:"#00A0DC", nextState:"turningOff"
+                attributeState "turningOff", label:'${name}', action:"switch.on", icon:"st.switches.light.off", backgroundColor:"#ffffff", nextState:"turningOn"
+            }
+        }
+        standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "default", label:"", action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+        main "switch"
+        details(["switch", "refresh"])
+    }
+}
+
+def installed() {
+         createChildDevices()
+         configure()
+}
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+    log.debug "description is $description"
+    Map map = zigbee.getEvent(description)
+    if (map) {
+        if (description?.startsWith('on/off')) {
```

Comment:
So when we make the fix to report endpoints, it probably will be a slightly different form than this as we have to maintain backwards compatibility for existing integrations.  I would recommend writing this DTH with the slower work around where we send a read when we receive on/off messages.  I know it isn't ideal, and we can update when we make the change in the FW, but for the time being I think it is the best option.

Reply:
I get the hub firmware from HQ, the hub will report ep, by catchall: 0104 0006 01 0A 0140 00 D29F 00 00 0000 0A 01 00001001, So the DTH parse this message. U means the HQ provided firmware may not be official released?

Reply:
That's correct, it's not officially released. I agree with Zach that the DTH should implement the workaround and when the hub firmware updates it will stop sending the `on/off` messages so the workaround will no longer be triggered.